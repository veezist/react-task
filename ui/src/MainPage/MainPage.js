import React from 'react';
import {Header} from './Header/Header'
import FlexView from 'react-flexview/lib';
import {List} from './List/List';



export class MainPage extends React.Component{

    constructor(props){
        super(props);
        this.state={
            servers:[],
            searchedServers: [],
        };
        
        this.handleTextChange = this.handleTextChange.bind(this);
        this.fetchPosts=this.fetchPosts.bind(this);
    }
  

    async fetchPosts() {
        fetch(`http://localhost:4454/servers`)
          .then(response => response.json())
          .then(
            data =>
              this.setState({
                servers: data,
                  searchedServers: data,  
              })
          ) 
      }

    handleTextChange(searchedServers) {
        this.setState({
            searchedServers: searchedServers
        });
    }

    componentDidMount(){
        this.fetchPosts();
    }

    render(){
        
        return (
        <FlexView column style={{width:'60%', margin: 'auto'}}>
        <Header onTextChange={this.handleTextChange} servers={this.state.servers} searchedServers={this.state.searchedServers}/>
        <List  searchedServers={this.state.searchedServers} />
        </FlexView>
            
        )
    }
}
