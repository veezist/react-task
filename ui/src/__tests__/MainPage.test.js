import React from 'react';
import {shallow} from 'enzyme';
import {MainPage} from '../MainPage/MainPage';





describe('<MainPage />',() => {
    it('renders main page', () => {
       const mainPage= shallow(<MainPage/>);
        expect(mainPage).toMatchSnapshot();
    });
});
