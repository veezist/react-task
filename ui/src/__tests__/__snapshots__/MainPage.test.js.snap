// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> renders main page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Header
          onTextChange={[Function]}
          searchedServers={Array []}
          servers={Array []}
        />,
        <List
          searchedServers={Array []}
        />,
      ],
      "column": true,
      "style": Object {
        "margin": "auto",
        "width": "60%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onTextChange": [Function],
          "searchedServers": Array [],
          "servers": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "searchedServers": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header
            onTextChange={[Function]}
            searchedServers={Array []}
            servers={Array []}
          />,
          <List
            searchedServers={Array []}
          />,
        ],
        "column": true,
        "style": Object {
          "margin": "auto",
          "width": "60%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onTextChange": [Function],
            "searchedServers": Array [],
            "servers": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "searchedServers": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
